jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -0,0 +1,6134 @@
+	 * @param size requested size of memory in bytes
+	 * @return starting address of memory
+	 * 
+	 * @throws OutOfMemoryError if no memory can be allocated
+	 * @throws IllegalArgumentException if size is not valid
+	 */
+	public long reallocateMemory(long address, long size) {
+		reallocateMemoryChecks(address, size);
+
+		if (0L == size) {
+			freeMemory(address);
+			return 0L;
+		}
+
+		long addressNew;
+		if (0L == address) {
+			addressNew = allocateMemory0(size);
+		} else {
+			addressNew = reallocateMemory0(address, size);
+		}
+
+		if (0L == addressNew) {
+			throw new OutOfMemoryError();
+		}
+
+		return addressNew;
+	}
+
+	/**
+	 * Set object at specified index in obj parameter
+	 * 
+	 * @param obj object into which to store the value
+	 * @param startIndex position of the value in obj
+	 * @param size the number of bytes to be set to the value
+	 * @param replace overwrite memory with this value
+	 *            
+	 * @throws IllegalArgumentException if startIndex is illegal in obj, or if size is invalid
+	 */
+	public void setMemory(Object obj, long startIndex, long size, byte replace) {
+		setMemoryChecks(obj, startIndex, size, replace);
+
+		if (0 != size) {
+			setMemory0(obj, startIndex, size, replace);
+		}
+	}
+
+	/**
+	 * Set object at specified index in parameter
+	 * 
+	 * @param startAddress specifies where to store value in memory
+	 * @param size the number of bytes to be set to the value
+	 * @param replace overwrite memory with this value
+	 *            
+	 * @throws IllegalArgumentException if startIndex is illegal, or if size is invalid
+	 */
+	public void setMemory(long startAddress, long size, byte replace) {
+		setMemory(null, startAddress, size, replace);
+	}
+
+	/**
+	 * Copy specified bytes from source object to destination
+	 * 
+	 * @param srcObj object to copy from 
+	 * @param srcOffset location in srcObj to start copy 
+	 * @param destObj object to copy into
+	 * @param destOffset location in destObj to start copy
+	 * @param size the number of bytes to be copied
+	 * 
+	 * @throws IllegalArgumentException if srcOffset is illegal in srcObj, 
+	 * if destOffset is illegal in destObj, or if size is invalid
+	 */
+	public void copyMemory(Object srcObj, long srcOffset, Object destObj, long destOffset, long size) {
+		copyMemoryChecks(srcObj, srcOffset, destObj, destOffset, size);
+
+		if (0 != size) {
+			copyMemory0(srcObj, srcOffset, destObj, destOffset, size);
+		}
+	}
+
+	/**
+	 * Copy specified bytes from source address to destination
+	 * 
+	 * @param srcAddress address to start copy
+	 * @param destAddress address to start copy
+	 * @param size the number of bytes to be copied
+	 * 
+	 * @throws IllegalArgumentException if srcAddress or destAddress
+	 *  is illegal, or if size is invalid
+	 */
+	public void copyMemory(long srcAddress, long destAddress, long size) {
+		copyMemory(null, srcAddress, null, destAddress, size);
+	}
+
+	/**
+	 * Swap specified bytes between source object and destination
```

Comment:
What happens when srcSize and destSize are different?

Reply:
Currently nothing. The `copySwapMemory0` native is only a stub implementation so there isn't much to go on.

Reply:
Hmm, I wonder if the javadoc should mention the method is currently unimplemented and will assert if called. @DanHeidinga thoughts?